Sources/gg.datagram.web-requests.sdPlugin/app.js

```diff
@@ -97,9 +97,20 @@ function sendWebSocket(data) {
     ws.onclose = function(evt) { onClose(this, evt); };
     ws.onopen = function() {
         onOpen(this);
+        const start = performance.now();
         ws.send(body);
-        ws.close();
-        showOk(data.context);
+        const readyCloseInterval = setInterval(function() {
+            if (ws.bufferedAmount == 0) {
+                ws.close();
+                showOk(data.context);
+                clearInterval(readyCloseInterval);
+            }
+            else if ((performance.now() - start) > 3000) {
+                showAlert(data.context);
```

Comment:
Oh, shouldn't we also close the connection when this happens? Otherwise I imagine this would end up being a connection leak.

Reply:
Agreed - PR updated with this